File: nicktelsan/day02/advancedPosition.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"strconv"
	"strings"
)

func main() {

	content, err := ioutil.ReadFile("day02/input")

	if err != nil {
		log.Fatal(err)
	}

	inputArray := strings.Split(string(content), "\n")

	const FORWARD = "forward"
	const UP = "up"
	const DOWN = "down"

	var horizontal = 0
	var depth = 0
	var aim = 0

	for i := 0; i < len(inputArray); i++ {
		splitValue := strings.Split(string(inputArray[i]), " ")
		value, _ := strconv.Atoi(splitValue[1])

		switch splitValue[0] {
		case FORWARD:
			horizontal += value
			depth += aim * value
		case UP:
			aim -= value
		case DOWN:
			aim += value
		}
	}

	distance := horizontal * depth
	fmt.Printf("Forward: %v\n", horizontal)
	fmt.Printf("Depth: %v\n", depth)
	fmt.Printf("Aim: %v\n", aim)
	fmt.Printf("Distance: %v\n", distance)
}

File: nicktelsan/day02/advancedPosition.js
import fs from 'fs'
import path from 'path'

const __dirname = path.resolve()

function main() {
  const input = fs.readFileSync(path.join(__dirname, 'day02/input'), 'utf8')
  const inputArray = input.split('\n')

  let horizontal = 0
  let depth = 0
  let aim = 0

  const FORWARD = 'forward'
  const UP = 'up'
  const DOWN = 'down'

  inputArray.forEach(value => {
    const splitValue = value.split(' ')
    switch (splitValue[0]) {
      case FORWARD:
        horizontal += parseInt(splitValue[1])
        depth += aim * parseInt(splitValue[1])
        break
      case UP:
        aim -= parseInt(splitValue[1])
        break
      case DOWN:
        aim += parseInt(splitValue[1])
    }
  })

  console.log(`Forward: ${horizontal}`)
  console.log(`Depth: ${depth}`)
  console.log(`Aim: ${aim}`)
  console.log(`Distance: ${horizontal * depth}`)
}

main()
